### PR TITLE
Make BMO e2e prow job required

### PR DESCRIPTION
**Pull request by lentzi90**
This removes the required status context for BMO e2e tests and instead makes the prow job required.

It also removes the report template because this is already covered by the report template set for plank. This config is currently causing double messages when a job fails (one for jenkins and one for plank).

One thing to discuss, test and decide is how to handle these required jobs.
Other required jobs that we have run automatically when needed. The e2e jobs have traditionally been manually triggered only to save some resources. I'm not sure if a required but not automatically run job will show up in the status contexts. If not, we have to decide if we

1. Add required status contexts back for them, or
2. Run the jobs automatically. If we choose this we can also use config like "skip if only" to not run it on all PRs, which would be quite nice!

### CODE REVIEW

File: prow/manifests/overlays/metal3/config.yaml
Comment (lentzi90) [inline prow/manifests/overlays/metal3/config.yaml:536]: What do you think? Should we trigger it automatically instead but have 
```
skip_if_only_changed: '(((^|/)OWNERS)|(\.md))$'
```

Comment (tuminoid) [reply prow/manifests/overlays/metal3/config.yaml:536]: I think this is the way to go. We've spent quite a lot of time developing the e2e test to make BMO testing more standalone, and its rather fast and stable test, so why wouldn't we want to run it?

**Review by Rozzii — APPROVED** [review]
/approve

**Comment by lentzi90** [comment]
/hold
@Rozzii what do you think about changing it to use `skip_if_only_changed`?

**Review by tuminoid — COMMENTED** [review]
/approve

**Comment by Rozzii** [comment]
> /hold @Rozzii what do you think about changing it to use `skip_if_only_changed`?

If the regexp works and it correctly skips the md and the OWNER files then I am fine with it.

**Comment by lentzi90** [comment]
After offline discussion, I think we can keep the old behavior of manual trigger but required job.
After merging this I will check if we need the required status check or not and do a follow up PR if needed.
/hold cancel

**Pull request merged**
